src/cls/IPM/CLI.cls

```diff
@@ -438,22 +438,44 @@ ClassMethod %ParseCommandInput(pCommandString As %String, Output pCommandInfo) A
 					Set tAccum = ""
 					Set tState = $$$PREARGUMENT
 				} ElseIf (tState = $$$MODIFIERNAME) {
+					set modifierSubscript = pCommandInfo
 					If $Data(tCommandStructure(pCommandInfo,"modifiers",tAccum)) {
 						Set tModifier = tAccum
 					} ElseIf $Data(tCommandStructure(pCommandInfo,"modifiers",1,tAccum),tAliasFor) {
 						Set tModifier = tAliasFor
+					} ElseIf (($data(pCommandInfo("parameters","actions"),phases)) && (pCommandInfo = "module-action")) || ($listfind(##class(%IPM.Main).#STANDARDPHASES,pCommandInfo)) {
+						
+						// Lifecycle phases can be called directly or through module-action, we need to share modifiers
+						if (pCommandInfo = "module-action") {
+							set phases = $listfromString(phases)
+						} else {
+							set phases = $lb("module-action")
+						}
+						set llen = $LISTLENGTH(phases)
```

Comment:
Nitpick: always use [$listnext](https://docs.intersystems.com/iris20243/csp/docbook/Doc.View.cls?KEY=RCOS_flistnext#RCOS_flistnext_listnext_and_performance) to iterate over $listbuild lists.